crates/simd-test-macro/src/lib.rs

```diff
@@ -60,19 +60,8 @@ pub fn simd_test(
         name.clone().as_str()
     ));
 
-    let default_target = if cfg!(target_os = "windows") {
-        Some("x86_64-pc-windows-msvc")
-    } else if cfg!(target_os = "linux") {
-        Some("x86_64-unknown-linux-gnu")
-    } else if cfg!(target_os = "macos") {
-        Some("x86_64-apple-darwin")
-    } else {
-        None
-    };
-
-    let target = env::var("TARGET").unwrap_or_else(|_| {
-        default_target.expect("TARGET environment variable not set and no default target known for the current target.").to_string()
-    });
+    let target = env::var("TARGET")
+        .expect("TARGET environment variable should be set for rustc");
     let mut force_test = false;
```

Comment:
So @alexcrichton merge as you wish but before these bits would set "TARGET" to sensitive defaults on all major platforms and otherwise complain.

How is just complaining any better?

Reply:
I mean, having to specify target for ci/run.sh is painful enough, if anything we should attempt to automatically detect target better.

That is, when someone writes `cargo test` things should just work.

Reply:
Oh that's what this PR is doing, automatically detecting `TARGET`! With this PR a bare `cargo test` should work.

The `TARGET` env var here in the procedural macro is carried over from the build script

Reply:
Ah nice, then let's merge this!